src/dvc_objects/fs/implementations/gdrive.py

```diff
@@ -51,172 +51,75 @@ def __init__(self, **config):
+        client_user_email = config.get("gdrive_service_account_user_email")
+        profile = config.get("profile")
 
-    def unstrip_protocol(self, path):
-        return f"gdrive://{path}"
+        if use_service_account:
+            client_json_file_path = config.get(
+                "gdrive_service_account_json_file_path"
+            )
+        else:
+            client_json_file_path = config.get("gdrive_user_credentials_file")
 
-    @staticmethod
-    def _get_kwargs_from_urls(urlpath):
-        return {"url": urlpath}
+        env_creds = os.getenv(self.GDRIVE_CREDENTIALS_DATA)
 
-    def _validate_config(self):
         # Validate Service Account configuration
-        if (
-            self._use_service_account
-            and not self._service_account_json_file_path
-            and GDriveFileSystem.GDRIVE_CREDENTIALS_DATA not in os.environ
-        ):
+        if use_service_account and not client_json_file_path and env_creds:
             raise ConfigError(
                 "To use service account, set "
                 "`gdrive_service_account_json_file_path`, and optionally"
                 "`gdrive_service_account_user_email` in DVC config."
             )
 
         # Validate OAuth 2.0 Client ID configuration
-        if not self._use_service_account:
-            if bool(self._client_id) != bool(self._client_secret):
+        if not use_service_account:
+            if bool(client_id) != bool(client_secret):
                 raise ConfigError(
                     "Please specify GDrive's client ID and secret in "
                     "DVC config or omit both to use the defaults."
                 )
 
-    @cached_property
-    def credentials_location(self):
-        """
-        Helper to determine where will GDrive remote read credentials from.
-        Useful for tests, exception messages, etc. Returns either env variable
-        name if it's set or actual path to the credentials file.
-        """
-        if os.getenv(GDriveFileSystem.GDRIVE_CREDENTIALS_DATA):
-            return GDriveFileSystem.GDRIVE_CREDENTIALS_DATA
-        if os.path.exists(self._creds_path):
-            return self._creds_path
-        return None
+        self._settings = {
+            "trash_only": config.get("gdrive_trash_only"),
+            "use_service_account": use_service_account,
+            "client_user_email": client_user_email,
+            "client_json_file_path": client_json_file_path,
+            "client_json": env_creds,
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "profile": profile,
+        }
+
+    @classmethod
+    def _strip_protocol(cls, path):
+        from fsspec.utils import infer_storage_options
+
+        opts = infer_storage_options(path)
+
+        return "{host}{path}".format(**opts)
+
+    def unstrip_protocol(self, path):
+        return f"gdrive://{path}"
 
     @staticmethod
-    def _validate_credentials(auth, settings):
```

Comment:
Q: do we have something similar in FS? Probably not critical but we are removing some safeguards (that we are missing in DVC). I don't remember the context, but probably it was coming from some confusion or something ...

Reply:
As I understand, this safeguard was mostly useful for checking that you are not accidentally using bad `.dvc/tmp/gdrive-user-credentails.json`. We now rely on per-client_id caching, so this situation shouldn't happen unless you are doing something fishy. Obviosly this could still happen if you are using your own custom keyfile location and messing this up, but that should be a very rare case now. Still, checking for cache and client_id/client_secret match would be great also for env creds (which we didn't have before and which seems like a more probable case), so we could add this in pydrive2.fs (will try to get back to this ASAP).

Reply:
no worries, I don't think that's a big deal